diff --git a/app/src/main/cpp/recipes/xkbcomp.cmake b/app/src/main/cpp/recipes/xkbcomp.cmake
index a7a2a33..3ef5bac 100644
--- a/app/src/main/cpp/recipes/xkbcomp.cmake
+++ b/app/src/main/cpp/recipes/xkbcomp.cmake
@@ -58,14 +58,17 @@ target_include_directories(xkbcomp
         PRIVATE
         "xkbcomp"
         "libx11/include"
-        "libx11/include/X11"
         "libx11/src"
         "libx11/src/xlibi18n"
         "libxkbfile/include/X11/extensions"
         "${CMAKE_CURRENT_BINARY_DIR}")
 target_link_libraries(xkbcomp PRIVATE xorgproto)
 target_link_options(xkbcomp PRIVATE "-fPIE" "-fPIC")
-target_compile_options(xkbcomp PRIVATE ${common_compile_options} "-fvisibility=hidden" "-DHAVE_STRCASECMP" "-DHAVE_STRDUP" "-DDFLT_XKB_CONFIG_ROOT=\"/\"" "-DHAVE_SYS_IOCTL_H" "-fPIE" "-fPIC" "-DPACKAGE_VERSION=\"2.70\"" "-Wno-shadow")
+target_compile_options(xkbcomp PRIVATE ${common_compile_options}
+        "-iquote${CMAKE_CURRENT_SOURCE_DIR}/libx11/include/X11"
+        "-fvisibility=hidden" "-DHAVE_STRCASECMP" "-DHAVE_STRDUP"
+        "-DDFLT_XKB_CONFIG_ROOT=\"/\"" "-DHAVE_SYS_IOCTL_H" "-fPIE"
+        "-fPIC" "-DPACKAGE_VERSION=\"2.70\"" "-Wno-shadow")
 target_apply_patch(xkbcomp "${CMAKE_CURRENT_SOURCE_DIR}/xkbcomp" "${CMAKE_CURRENT_SOURCE_DIR}/patches/xkbcomp.patch")
 target_apply_patch(xkbfile "${CMAKE_CURRENT_SOURCE_DIR}/libxkbfile" "${CMAKE_CURRENT_SOURCE_DIR}/patches/xkbfile.patch")
 target_apply_patch(X11 "${CMAKE_CURRENT_SOURCE_DIR}/libx11" "${CMAKE_CURRENT_SOURCE_DIR}/patches/x11.patch")
